sympy/functions/combinatorial/factorials.py

```diff
@@ -835,7 +835,11 @@ class binomial(CombinatorialFunction):
     The function is also generalized to handle negative values of
     ``k``.
 
-    XXX add mathematical definition for this case
+    .. math:: \binom{n}{k} = \binom{n}{n - k}
+
+    If both `n` and `k` are negative integers, and `n` is greater than `k`.
```

Comment:
That looks too simple :-) What if n and k are not integers and neither n not n - k is an integer?
```
>>> binomial(-2.5,-1.3)
0.0938281251419306
>>> binomial(-2.5,1.3)
-1.77900864608809
```

Reply:
then the existing implementation holds ... 
the `gamma` part is invoked, it doesn't need anything extra
right ?

Reply:
I mean the existing implementation of binomial ... works like this
```python3
>>> binomial(-3.0001,-5.0001)
6.00035000497335
>>> binomial(-3,-5)
0
```
but ours is 
```python3
>>> binomial(-3,-5.0)
6
>>> binomial(-3,-5)
6
```

Reply:
> That looks too simple :-) What if n and k are not integers and neither n not n - k is an integer?
> 
> ```
> >>> binomial(-2.5,-1.3)
> 0.0938281251419306
> >>> binomial(-2.5,1.3)
> -1.77900864608809
> ```

How do you suggest we introduce this ?

Reply:
See latest commit which is on its way.